### PR TITLE
[Merged by Bors] - Add GOPRIVATE variable in CI workflow

### PR DESCRIPTION
**Pull request by dsmello**
## Motivation

Add the support to Go Private modules on the CI pipeline and manage them using the Repository variables.

## Description
Allow the tests and usage of private repositories.

### CODE REVIEW

**Comment by fasmat** [comment]
Does this also work on Windows/Mac?

Also the other workflows like systests, release, etc. need to be updated as well 😊

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,6 +14,7 @@ env:
   TD_QUEUE_ZONE: ${{ secrets.TD_QUEUE_ZONE }}
   DUMP_QUEUE_NAME: ${{ secrets.DUMP_QUEUE_NAME }}
   DUMP_QUEUE_ZONE: ${{ secrets.DUMP_QUEUE_ZONE }}
+  # If was necessary the variable GOPRIVATE can be set on the repository settings :: Secrets and Variables > action > Variables : GOPRIVATE
```
Comment (fasmat) [inline .github/workflows/ci.yml]:
```suggestion
  # If needed the variable GOPRIVATE can be set in the repository settings :: Secrets and Variables > action > Variables : GOPRIVATE
```

I assume the value of the variable has to be what `go get` expects, i.e. the url of the private repo(s)?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -91,6 +99,13 @@ jobs:
         uses: actions/checkout@v4
         with:
           ssh-key: ${{ secrets.GH_ACTION_PRIVATE_KEY }}
+
+      - uses: extractions/netrc@v1
+        with:
+          machine: github.com
+          username: ${{ secrets.GH_ACTION_TOKEN_USER }}
+          password: ${{ secrets.GH_ACTION_TOKEN }}
+        if: vars.GOPRIVATE
```
Comment (fasmat) [inline .github/workflows/ci.yml:108]:
```suggestion
        if: ${{ vars.GOPRIVATE }}
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -128,6 +143,13 @@ jobs:
         uses: actions/checkout@v4
         with:
           ssh-key: ${{ secrets.GH_ACTION_PRIVATE_KEY }}
+      
+      - uses: extractions/netrc@v1
+        with:
+          machine: github.com
+          username: ${{ secrets.GH_ACTION_TOKEN_USER }}
+          password: ${{ secrets.GH_ACTION_TOKEN }}
+        if: vars.GOPRIVATE
```
Comment (fasmat) [inline .github/workflows/ci.yml:152]:
```suggestion
        if: ${{ vars.GOPRIVATE }}
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -167,6 +189,13 @@ jobs:
...
```
Comment (fasmat) [inline .github/workflows/ci.yml:198]:
```suggestion
        if: ${{ vars.GOPRIVATE }}
```

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -219,6 +248,13 @@ jobs:
         with:
           lfs: true
           ssh-key: ${{ secrets.GH_ACTION_PRIVATE_KEY }}
+      
+      - uses: extractions/netrc@v1
+        with:
+          machine: github.com
+          username: ${{ secrets.GH_ACTION_TOKEN_USER }}
+          password: ${{ secrets.GH_ACTION_TOKEN }}
+        if: vars.GOPRIVATE
```
Comment (fasmat) [inline .github/workflows/ci.yml:257]:
```suggestion
        if: ${{ vars.GOPRIVATE }}
```

**Review by fasmat — CHANGES_REQUESTED** [review]

**Review by andres-spacemesh — APPROVED** [review]

**Review by andres-spacemesh — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (dsmello) [reply .github/workflows/ci.yml:108]: The "if" field didn't requires the template brackets

**Comment by dsmello** [comment]
> Does this also work on Windows/Mac?
> 
> Also the other workflows like systests, release, etc. need to be updated as well 😊

Mac and Windows :: Yes, its suppose to work in all systems.

**Comment by dsmello** [comment]
I'll apply the same changes on the others pipelines

File: .github/workflows/ci.yml (hunk 2 above)
Comment (fasmat) [reply .github/workflows/ci.yml:108]: OK; just for consistency the first use of the variable (line 76) uses template brackets - I'm also OK with removing them there - just to have the same in every job 🙂

**Review by fasmat — APPROVED** [review]

**Comment by fasmat** [comment]
bors merge

**Pull request closed without merging**